src/components/Sidebar.js

```diff
@@ -8,35 +8,22 @@ const StyledSidebar = styled.div`
   justify-content: center;
   align-items: center;
 `
+const getAllFileNames = () => {
+  return fetch('data/trails/all-trails.json');
+};
 
-// Accepts a directory path as a parameter
-// Returns a list of all files in that directory
-const getFileList = (dirPath) => {
-  return fs.readdirSync(dirPath)
-}
-
-// Accepts a JSON's file path as a parameter
-// Returns the contents of that JSON file
-const getFileContent = (filePath) => {
-  return require(filePath);
-}
-
-// Accepts a directory path as a parameter
-// Returns an object with the keys as each file name
-// within that directory and the values as 
-// the contents of that file. 
-// Expects each file to be a JSON file.
-const getAllFileContents = async (dirPath) => {
-  const data = {}
-  const files = fs.readdirSync(dirPath);
-  for (const file of files) {
-    const fileContent = require(`${dirPath}/${file}`)
-    data[file] = fileContent;
-  }
-  return data;
-}
 
 export default function Sidebar() {
+  useEffect(() => {
+    getAllFileNames()
+    .then(response => response.json())
+    .then(response => {
+      for (const trail of response.trails) {
```

Comment:
Can also use `trails.forEach(trail => ...`

Nothing wrong with your approach, I just find the forEach to be more commonly used than for (blah of blahs)